In [2]:
import psycopg2
import numpy as np
import matplotlib.pyplot as plt
from utils.Database_Connection import DatabaseConnection
from sequence_processor import SequenceProcessor

conn = DatabaseConnection(
    host='localhost',
    database='cadets-e3-local',
    user='studi',
    password='studi',
    port=7654
)
processor = SequenceProcessor(conn)
processor.process_sequences()

print(f"Statistiken:")
print(f"Anzahl Zeilen: {processor.stats['total_rows']}")
print(f"Anzahl Sequenzen: {processor.stats['total_sequences']}")
print(f"Sequenzen nach Subject: {processor.stats['sequences_by_subject']}")
print(f"Sequenzen nach Executable: {processor.stats['sequences_by_executable']}")
print(f"Durchschnittliche Länge: {processor.stats['total_length'] / processor.stats['total_sequences']:.2f}")

Statistiken:
Anzahl Zeilen: 41291622
Anzahl Sequenzen: 431895
Sequenzen nach Subject: 224146
Sequenzen nach Executable: 207749
Durchschnittliche Länge: 95.60



### Sequence Processor
- in diesem ersten Teil werden die Exucatables aus den Prozessen gezogen und es wird für jedes Executable ein Eintrag angelegt


### Anmerkungen
- ich habe im Gegensatz zu jannik ein primary key erstellt, um die eindeutigkeit der sequenzen zu gewährleisten
- Warum habe ich weniger Zeilen als Jannik?
    - generell habe ich ca 2000000 events und jannik ca 40000000
    - Habe ich villeicht nicht alle Daten kopiert in meinen Docker Container?
    - Hat das initial_clenaup.sql viele Daten gelöscht?
    - Ergebniss von Jannik:
        - count_rows: 41291622
        - count_sequences: 431896
        - count_sequences_by_subject: 224147
        - count_sequences_by_executable: 207749
        - avg length: 95.6036846833497
- Es ist wichtig zu beachten, das ich nur die Daten zum Training verwende, in denen keine Angriffe stattgefunden haben. Dies ist notwendig, um sicherzustellen, dass die Trainingsdaten nur gutartige Daten enthalten. 
    - der erste Angriff hat am 2018-04-06 um 11:00 Uhr stattgefunden
- Wichtig ist zu schauen, das beim filtern der einzelnen Sequenzen im Prozess nicht nach dem executable gruppiert wird, da sonst 2 eigentlich unterschiedliche Sequenzen zu einer Sequenz zusammengefasst werden, wenn sie das gleiche executable haben.
    - dieses Problem wird gelöst, indem die sequenzen nach der sequence_long geordnet werden, diese gibt nämlich die Reihenfolge der events an

    

In [2]:
from SequenceEventProcessor import SequenceEventProcessor
from utils.Database_Connection import DatabaseConnection


conn = DatabaseConnection(
    host='localhost',
    database='cadets-e3-local',
    user='studi',
    password='studi',
    port=7654
)

events_processor = SequenceEventProcessor(conn)
events_processor.process_sequence_events()

print(events_processor.stats)

{'total_events': 21307067, 'total_sequences': 131603}


In [1]:
from Sequence_Exporter import SequenceDataExporter
from utils.Database_Connection import DatabaseConnection


conn2 = DatabaseConnection(
    host='localhost',
    database='cadets-e3-local',
    user='studi',
    password='studi',
    port=7654
)

exporter = SequenceDataExporter(conn2)
exporter.export_sequences("data.npy", "targets.npy")



['EVENT_CLOSE', 'EVENT_OPEN', 'EVENT_CLOSE', 'EVENT_EXECUTE', 'EVENT_OPEN', 'EVENT_READ', 'EVENT_CLOSE', 'EVENT_OPEN', 'EVENT_READ', 'EVENT_READ', 'EVENT_CLOSE', 'EVENT_OPEN', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_CLOSE', 'EVENT_OPEN', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_CLOSE', 'EVENT_OPEN', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_CLOSE', 'EVENT_OPEN', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_CLOSE', 'EVENT_OPEN', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_CLOSE', 'EVENT_OPEN', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_CLOSE', 'EVENT_OPEN', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_MMAP', 'EVENT_CLOSE', 'EVENT_OPEN', 'EVENT_MMAP', 'EVENT_MMAP', 'EVEN